@xen-orchestra/fs/src/abstract.js

```diff
@@ -283,6 +283,27 @@ export default class RemoteHandlerAbstract {
   async read(file: File, buffer: Buffer, position?: number): Promise<{| bytesRead: number, buffer: Buffer |}> {
     return this._read(typeof file === 'string' ? normalizePath(file) : file, buffer, position)
   }
+  /**
+   * Copy a range from one file to the other, kernel side, server side or with a reflink if possible.
+   *
+   * Slightly different from the copy_file_range linux system call:
+   *  - offsets are mandatory (because some remote handlers don't have a current pointer for files)
+   *  - flags is fixed to 0
+   *  - will not return until copy is finished.
+   *
+   * @param fdIn read open file descriptor
+   * @param offsetIn either start offset in the source file
+   * @param fdOut write open file descriptor (not append!)
+   * @param offsetOut offset in the target file
+   * @param dataLen how long to copy
+   * @returns {Promise<void>}
+   */
+  async copyFileRange(fdIn, offsetIn, fdOut, offsetOut, dataLen) {
+    // default implementation goes through the network
+    const buffer = Buffer.alloc(dataLen)
+    await this._read(fdIn, buffer, offsetIn)
+    await this._write(fdOut, buffer, offsetOut)
+  }
 
   async readFile(file: string, { flags = 'r' }: { flags?: string } = {}): Promise<Buffer> {
```

Comment:
Method in `Abstract` should only be small wrappers around implementations (`_`).

They should handle high-level features like path normalization.

Reply:
what change do you suggest I make?

Reply:
https://github.com/vatesfr/xen-orchestra/blob/f823690b44d8eda47a26e754e0424b67e0a96b4a/%40xen-orchestra/fs/src/abstract.js#L306-L308